### PR TITLE
Improve contributing

### LINKED ISSUE #36: Refactor CONTRIBUTING
We need to explain better how to contribute to the project, a good guidelines about which commands contributors will need and the purpose, how to work with Lerna and packages (alltogether or separately), how to create and add new packages, how to flat devDependencies and how we manage dependencies. 

Also, we think this project is a good point to start trying the new VSCode feature called [Remote Development](https://code.visualstudio.com/docs/remote/remote-overview) to give new contributors a environment with some great settings and plugins they would need to develop, but this will be done before and improved in next iterations with more features

Things to do:
- [x] Refactor Contributor guide
- [x] Explain Lerna + Yarn Workspaces and how we use them
- [x] Explain useful Lerna commands
- [x] Explain how to work with packages alltogether (example, how to run tests or lint before pushing to git)
- [x] Explain how to work with packages in their simple unit (example, how to run compiler to see if my new feature in package x compiles)
- [x] Explain how to create packages
- [x] Explain how to import packages
- [x] Explain how we manage dependencies, devDependencies and how to flatten devDeps
- [x] _(Optionally)_ Explain how to work with VSCode Remote Development

### PR DESCRIPTION
**Pull request by sergiohgz**
**Type:** refactor
**Scope:** root

The following has been addressed in the PR:

*  There is a related issue? #36
*  Unit or Functional tests are included in the PR **No**

**Description:** Recreate the Contributing guide to let users know how the project is built, how it works and how to work.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,32 +1,314 @@
...
+
+The steps for importing and creating packages differ a little bit. First, you need to have cloned the project to import in your local filesystem and then you need to import all the project git tree using next command:
+
+`lerna import <project_location> --dest=<scope_location>`
+
+If this process fails because there were merge conflicts, you must add `--flatten` option to the command above to flat history.
+
+Later, follow the next steps in the package recently imported:
+
+1. Remove unnecessary stuff, like CI settings, GitHub stuff, CONTRIBUTING, unnecessary things from .gitignore, devDependencies, scripts, etc.
```
Comment (griffithtp) [inline CONTRIBUTING.md]: What do you think about providing a checklist of what should be removed, something like a specific checklist now that we have them in the root folder? Something like,
```
Review the list of files you are importing, notably:
* .git folder - should be removed
* .gitignore - please review against the main .gitignore file
* remove any build pipeline files you may have
* update devDependencies and Dependencies to hoist packages from the main package.json
* update scripts
```

Comment (sergiohgz) [reply CONTRIBUTING.md]: Nice, I will elaborate a complete checklist for each step to not leave things in the air :wink:

**Comment by sergiohgz** [comment]
@griffithtp @juanpicado, this is ready for review and merge. I'm creating a series of article for the Verdaccio Blog where I will explain the purpose, how to create a monorepo, tools, and more :wink:

**Comment by juanpicado** [comment]
Thanks Serio I will check it out :)

**Pull request merged**